generic/lshw.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM.
+# Author: Ramya BS <ramya@linux.vnet.ibm.com>
+
+import os
+import re
+import subprocess
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Lshwrun(Test):
+
+    """
+    lshw is a small tool to extract detailed information on the
+    hardware configuration of the machine.
+    It can report exact memory configuration,firmware version,
+    mainboard configuration, CPU version and speed,cache configuration, bus
+    speed, etc. on DMI-capable x86 or IA-64 systems and on some PowerPC
+    machines (PowerMac G4 is known to work).
+    """
+    interface = process.system_output("ip route show")
+    active_interface = re.search(
+        r"default via\s+\S+\s+dev\s+(\w+)\s+proto", interface).group(1)
```

Comment:
This can be added to avocado utils, since others could use it as well. Please consider